internal/core/services/operation_manager/operation_manager.go

```diff
@@ -331,3 +263,37 @@ func (om *OperationManager) StartLeaseRenewGoRoutine(operationCtx context.Contex
 		}
 	}()
 }
+
+func (om *OperationManager) addOperationsLeaseData(ctx context.Context, schedulerName string, ops []*operation.Operation) error {
+	for _, op := range ops {
+		ttl, err := om.leaseStorage.FetchLeaseTTL(ctx, schedulerName, op.ID)
```

Comment:
Something that crossed my mind is that what we want for the API is to list all operations.
So we go to redis once, get all operations, then go back to the server and sort it in lists.
What we added makes us go to redis n times (being n the number of operations on redis). Since we have to list all operations anyways, do you think it would be bad to have a method that get all operation leases, so that we can iterate through them changing those 1+n redis queries to only 2 (all operations, then all leases)?

Reply:
It wouldn't be bad if we had this method to return all operation leases, but I question if this is an actual necessity for the current model. You can see that this method iterates over all active operations **for a given scheduler**, and in the current model (one worker model) this number will always be 1, so I'm not sure if we need this optimization now. Even with the multiple workers model, the active status is always a temporary status and not a final one (like StatusFinished for example), this means that this number does not scale as the finished operations do. Anyway, we are still querying all operations (finished, pending and active) in the storage methods, since we are using pagination, it would be nice to use some kind of "LIMIT" at least in the finished operations query (since it can get really huge). I'm going to search if we can limit those queries using redis.